src/library/themes/__tests__/themeFromTokens.spec.js

```diff
@@ -0,0 +1,144 @@
+  color_brand: 'blue'
+};
+const TOKENS_DANGER = {
+  color_danger: 'red'
+};
+const TOKENS_DEFAULT = {
+  color_theme: 'blue',
+  pixel: '16px',
+  boxShadow: '1px 1px 1px rgba(0,0,0,0.1)',
+  breakpoint: '16px',
+  fontSize_test: '1rem',
+  untouched: 100
+};
+const TOKENS_GRAY = {
+  color_gray: 'gray'
+};
+const TOKENS_WHITE = {
+  color_white: 'white'
+};
+
+const themeFromColorTokensWithAliases = (color, tokens) =>
+  themeFromColorTokens({
+    aliases: ALIASES,
+    color,
+    colorName: 'unimportant',
+    tokens
+  });
+
+describe('themeFromTokens', () => {
+  describe('converts', () => {
+    const theme = themeFromTokens(TOKENS_DEFAULT);
+    it('rems to ems', () => {
+      expect(theme.fontSize_test).toMatchSnapshot();
+    });
+
+    it(`'brand' to 'theme' in the key`, () => {
+      expect(theme.color_theme).toMatchSnapshot();
+      expect(theme.color_brand).toBeUndefined();
+    });
+
+    it('leaves some stuff alone', () => {
+      expect(theme.boxShadow).toMatchSnapshot();
+      expect(theme.untouched).toMatchSnapshot();
+    });
+
+    describe('px to ems', () => {
+      it('when not a breakpoint', () => {
+        expect(theme.pixel).toMatchSnapshot();
+      });
+
+      it(`except when key contains 'breakpoint'`, () => {
+        expect(theme.breakpoint).toMatchSnapshot();
+      });
+    });
+  });
+});
+
+describe('themeFromTokens', () => {
+  describe('with colors', () => {
+    describe('with black', () => {
+      it('overrides black value', () => {
+        const theme = themeFromColorTokensWithAliases(
+          'custom_black',
+          TOKENS_BLACK
+        );
+        expect(theme.color_black).toMatchSnapshot();
+      });
+    });
+
+    describe('with intent', () => {
+      it('overrides intent value', () => {
+        const theme = themeFromColorTokensWithAliases(
+          CUSTOM_RAMP,
+          TOKENS_DANGER
+        );
+        expect(theme.color_danger).toMatchSnapshot();
+      });
+    });
+
+    describe('with gray', () => {
+      it('overrides gray value', () => {
+        const theme = themeFromColorTokensWithAliases(CUSTOM_RAMP, TOKENS_GRAY);
+        expect(theme.color_gray).toMatchSnapshot();
+      });
+    });
+
+    describe('with theme', () => {
+      it('overrides theme value', () => {
+        const theme = themeFromColorTokensWithAliases(
+          CUSTOM_RAMP,
+          TOKENS_BRAND
+        );
+        expect(theme.color_theme).toMatchSnapshot();
+      });
+    });
+
+    describe('with white', () => {
+      it('overrides white value', () => {
+        const theme = themeFromColorTokensWithAliases(
+          'custom_white',
+          TOKENS_WHITE
+        );
+        expect(theme.color_white).toMatchSnapshot();
+      });
+    });
+
+    describe('with invalid ramp as any color', () => {
+      it('throws error', () => {
+        expect(() => {
+          themeFromColorTokensWithAliases(BAD_RAMP, TOKENS_DANGER);
+        }).toThrowError();
+      });
+    });
+  });
+});
```

Comment:
Great suite of tests- it's really clear what each of them is measuring. Curious if you explored what happens when someone tries to set a color to a string that doesn't exist in the palette? e.g. `color_danger: 'coral'`

Reply:
They'll get the [error thrown here](https://github.com/mineral-ui/mineral-ui/pull/691/commits/365b76a08ec873fdc43cbea704cfcffd626908e5#diff-50c3e93e648bfd3f62bed12ec8bade1aR72):

```
[mineral-ui/themes/createTheme]: Expected colors.danger to be a color ramp from the mineral-ui palette or a custom color ramp, but got 'coral'.
See https://mineral-ui.com/color#guidelines-ramps for the palette ramps.
See https://github.com/mineral-ui/mineral-ui/blob/master/packages/mineral-ui-tokens/src/blue.js for an example color ramp.
```

Reply:
Cool, that's what I thought, but wasn't sure if any of these tests covered that case.

Reply:
That would be here: https://github.com/mineral-ui/mineral-ui/pull/691/commits/869e73d898dea59904a7229098cc036517453c25#diff-82e2f718ed7884d04ebb1553ad8939e7R126